src/tree.rs

```diff
@@ -436,6 +438,63 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Swaps two `Node`s including their children given their `NodeId`s.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    ///
+    /// Notes:
+    ///     - Both `NodeId`s are still valid after this process and are not swapped.
+    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    ///
+    /// ``
```

Comment:
Alright, I did some testing.
Using `move_node_to_parent` two times actually works just nicely for moves *across*.

But for *down* and especially swapping the root break things spectacularly.
Lets assume the following example:
```
  A
 / \
B   C
|   |
D   E
```

`swap_sub_trees(D, E)` using `move_node_to_parent(E, D_Parent) ` and `move_node_to_parent(D, E_Parent) `
```
Step 1: (move_node_to_parent(E, D_Parent))
    A
   / \
  B   C
 / \
D   E

Step 2: (move_node_to_parent(D, E_Parent))
    A
   / \
  B   C
  |   |
  E   D
```

Swapping with the same parent also works with the correct order:
`swap_sub_trees(B, C)`
```
Step 1: (move_node_to_parent(C, B_Parent))
    A
   / \
  B   C
  |   |
  D   E

Step 2: (move_node_to_parent(B, C_Parent))
    A
   / \
  C   B
  |   |
  D   E
```

But breaks once we have more then two children.
This successfully swaps `B` and `C` but also moves `F` in the process:
`swap_sub_trees(B, C)`
```
     A
   / | \
  B  C  F
  |  |
  D  E

Step 1: (move_node_to_parent(C, B_Parent))
     A
   / | \
  B  F  C
  |     |
  D     E

Step 2: (move_node_to_parent(B, C_Parent))
     A
   / | \
  F  C  B
     |  |
     E  D
```

Lets get more interesting:
`swap_sub_trees(B, D)`
```
Step 1: (move_node_to_parent(D, B_Parent))
     A
   / | \
  B  C  D
     |
     E

Step 2: (move_node_to_parent(B, D_Parent))
This is the first time D_Parent is not entirely clearly defined.
If it is the parent at the start of the move this means move B under B (woot?).
If its the parent now we end up with:
     A
   / | \
  C  D  B
  |
  E

Expected:
    A
   / \
  C   D 
  |   |
  E   B
```
Not exactly what we wanted.

With the root involved this gets even more crazy:
`swap_sub_trees(A, D)`
```
Step 1: (move_node_to_parent(D, A_Parent))
A has no parent, does this mean we orphan D?
Lets just ignore this for now.

Step 2: (move_node_to_parent(A, D_Parent))
     B
     |
     D
     |
     A
     |
     C
     |
     E

Step 3: Maybe change the order and do  (move_node_to_parent(D, A_Parent)) now?
Still makes now sense. Either A_Parent is the original parent, which is still None,
or the Parent of A is A, so move_node_to_parent(A, A)??

Expected:
  D
 / \
B   C
|   |
A   E
```

I think this function needs a lot more logic then currently provided by `move_node_to_parent`, when we have a sub-tree match.
Even moves *across* the tree move is not perfect in terms of order (see the three children example), which leads me to believe even that should not be handled by any calls to `move_node_to_parent`.

If you have any more thoughts please tell me, I will most likely start with a clean new implementation now. My old one was to simple and your idea does not live up to our needs.